src/r/recordFactories/ProjectFactory.ts

```diff
@@ -419,6 +445,78 @@ export class ProjectFactory extends RecordFactory<RT.project> {
     }
   }
 
+  addLinkedVariables<N extends RT>(this: ProjectFactory, records: RecordNode<N>[]): void {
+    records.forEach(record => {
```

Comment:
Use for-of loop. Has better performance as it is called sync. 
forEach is called in async fashion with a callback and incurs a function call overhead.